cmd/ci-operator-configresolver/main.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/interrupts"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+)
+
+const (
+	orgQuery     = "org"
+	repoQuery    = "repo"
+	branchQuery  = "branch"
+	variantQuery = "variant"
+)
+
+type filenameToConfig map[string]api.ReleaseBuildConfiguration
+
+type options struct {
+	configs     filenameToConfig
+	configPath  string
+	logLevel    string
+	address     string
+	gracePeriod time.Duration
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.configPath, "config", "", "Path to config dirs")
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	fs.StringVar(&o.address, "address", ":8080", "Address to run server on")
+	fs.DurationVar(&o.gracePeriod, "gracePeriod", time.Second*10, "Grace period for server shutdown")
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func validateOptions(o options) error {
+	_, err := log.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %v", err)
+	}
+	return nil
+}
+
+func missingQuery(w http.ResponseWriter, field string) {
+	w.WriteHeader(http.StatusBadRequest)
+	fmt.Fprintf(w, "%s query missing or incorrect", field)
+	return
+}
+
+func resolveConfig(o *options) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusNotImplemented)
+			w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+		org := r.URL.Query().Get(orgQuery)
+		if org == "" {
+			missingQuery(w, orgQuery)
+			return
+		}
+		repo := r.URL.Query().Get(repoQuery)
+		if repo == "" {
+			missingQuery(w, repoQuery)
+			return
+		}
+		branch := r.URL.Query().Get(branchQuery)
+		if branch == "" {
+			missingQuery(w, branchQuery)
+			return
+		}
+		variant := r.URL.Query().Get(variantQuery)
+		info := config.Info{
+			Org:     org,
+			Repo:    repo,
+			Branch:  branch,
+			Variant: variant,
+		}
+		info.Filename = filepath.Join(o.configPath, branch, info.Basename())
```

Comment:
Since we have separate configmaps for each branch, they get mounted to their own subdirs. That being said, there is no reason to set the `info.Filename` field anymore. That was used for the earlier PR for this which loaded the config file dynamically as requested from clients. I'll just remove this as it's unnecessary. The way that we load configs now is directly from the `filenameToConfig` map.